File: client/src/app/modules/miscellaneous/services/guichet-unique/forme.juridique.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { FormeJuridique } from 'src/app/modules/quotation/model/guichet-unique/referentials/FormeJuridique';
import { AppRestService } from 'src/app/services/appRest.service';

@Injectable({
  providedIn: 'root'
})
export class FormeJuridiqueService extends AppRestService<FormeJuridique>{

  constructor(http: HttpClient) {
    super(http, 'quotation/guichet-unique');
  }

  getFormeJuridique() {
    return this.getList(new HttpParams(), 'forme-juridique');
  }

}                        

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/guichetUnique/referentials/FormeJuridiqueService.java
package com.jss.osiris.modules.osiris.quotation.service.guichetUnique.referentials;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.guichetUnique.referentials.FormeJuridique;

public interface FormeJuridiqueService {
    public List<FormeJuridique> getFormeJuridique();

    public FormeJuridique getFormeJuridique(String code);
}
